src/test/lib/matches_all_pos_list_test.cpp

```diff
@@ -0,0 +1,65 @@
+#include "base_test.hpp"
+#include "gtest/gtest.h"
+#include "operators/get_table.hpp"
+#include "operators/maintenance/create_table.hpp"
+#include "operators/table_wrapper.hpp"
+#include "storage/matches_all_pos_list.hpp"
+#include "storage/pos_list.hpp"
+
+namespace opossum {
+
+class MatchesAllPosListTest : public BaseTest {
+ public:
+  void SetUp() override {
+    column_definitions.emplace_back("a", DataType::Int, false);
+    column_definitions.emplace_back("b", DataType::Float, true);
+
+    dummy_table_wrapper = std::make_shared<TableWrapper>(Table::create_dummy_table(column_definitions));
+    dummy_table_wrapper->execute();
+
+    create_table = std::make_shared<CreateTable>("t", false, dummy_table_wrapper);
+  }
+
+  TableColumnDefinitions column_definitions;
+  std::shared_ptr<TableWrapper> dummy_table_wrapper;
+  std::shared_ptr<CreateTable> create_table;
+};
+
+TEST_F(MatchesAllPosListTest, AddAfterMatchedAllTest) {
```

Comment:
Pending the discussion above, this test (and the setup above) might be replaced with an `EXPECT_THROW`. However, I am missing a test that tests the intended use of this type of pos list.